packages/components/src/components/post-megadropdown/post-megadropdown.tsx

```diff
@@ -75,6 +77,9 @@ export class PostMegadropdown {
     PostMegadropdown.activeDropdown = this;
     this.postToggleMegadropdown.emit(this.isVisible);
     this.addOutsideClickListener();
+    this.triggerElement = document.querySelector(
```

Comment:
In theory there could be multiple triggers. 
Can you rather listen to the `postToggleMegadropdown` in the trigger that opened the megadropdown?